lua_modules/dht22/dht22.lua

```diff
@@ -0,0 +1,101 @@
+-- ***************************************************************************
+-- DHT22 module for ESP8266 with nodeMCU
+--
+-- Written by Javier Yanez 
+-- but based on a script of Pigs Fly from ESP8266.com forum
+--
+-- MIT license, http://opensource.org/licenses/MIT
+-- ***************************************************************************
+
+local moduleName = ...
+local M = {}
+_G[moduleName] = M
+
+local humidity
+local temperature
+local checksum
+local checksumTest
+
+function M.read(pin)
+  humidity = 0
+  temperature = 0
+  checksum = 0
+
+  -- Use Markus Gritsch trick to speed up read/write on GPIO
+  gpio_read = gpio.read
```

Comment:
It is a speed optimization. I do not understand at all, but this loop:

```javascript
while (gpio_read(pin) == 1 and c < 100) do c = c + 1 end
```
It is faster than this:

```javascript
while (gpio.read(pin) == 1 and c < 100) do c = c + 1 end
```